docs/field-level-encryption.md

```diff
@@ -0,0 +1,112 @@
+# Integrating with MongoDB [Client Side Field Level Encryption](https://www.mongodb.com/docs/manual/core/csfle/)
+
+Client Side Field Level Encryption, or CSFLE for short, is a tool for storing your data in an encrypted format in MongoDB.
+For example, instead of storing the `name` property as a plain-text string, CSFLE means MongoDB will store your document with `name` as an encrypted buffer.
+The resulting document will look similar to the following to a client that doesn't have access to decrypt the data.
+
+```
+{
+  "_id" : ObjectId("647a3207661e3a3a1bc3e614"),
+  "name" : BinData(6,"ASrIv7XfokKwiCUJEjckOdgCG+u6IqavcOWX8hINz29MLvcKDZ4nnjCnPFZG+0ftVxMdWgzu6Vdh7ys1uIK1WiaPN0SqpmmtL2rPoqT9gfhADpGDmI60+vm0bJepXNY1Gv0="),
+  "__v" : 0
+}
+```
+
+You can read more about CSFLE on the [MongoDB CSFLE documentation](https://www.mongodb.com/docs/manual/core/csfle/) and [this blog post about CSFLE in Node.js](https://www.mongodb.com/developer/languages/javascript/client-side-field-level-encryption-csfle-mongodb-node/).
+
+Note that Mongoose does **not** currently have any Mongoose-specific APIs for CSFLE.
+Mongoose defers all CSFLE-related work to the MongoDB Node.js driver, so the [`autoEncryption` option](https://mongodb.github.io/node-mongodb-native/5.6/interfaces/AutoEncryptionOptions.html) for `mongoose.connect()` and `mongoose.createConnection()` is where you put all CSFLE-related configuration.
+Mongoose schemas currently don't support CSFLE configuration.
+
+## Setting Up Field Level Encryption with Mongoose
+
+First, you need to install the [mongodb-client-encryption npm package](https://www.npmjs.com/package/mongodb-client-encryption).
+This is MongoDB's official package for setting up encryption keys.
+
+```
+npm install mongodb-client-encryption
+```
+
+You also need to make sure you've installed [mongocryptd](https://www.mongodb.com/docs/manual/core/queryable-encryption/reference/mongocryptd/).
+mongocryptd is a separate process from the MongoDB server that you need to run to work with field level encryption.
+You can either run mongocryptd yourself, or make sure it is on the system PATH and the MongoDB Node.js driver will run it for you.
+[You can read more about mongocryptd here](https://www.mongodb.com/docs/v5.0/reference/security-client-side-encryption-appendix/#mongocryptd).
+
+Once you've set up and run mongocryptd, first you need to create a new encryption key as follows.
+Keep in mind that the following example is a simple example to help you get started.
+The encryption key in the following example is insecure; MongoDB recommends using a [KMS](https://www.mongodb.com/docs/v5.0/core/security-client-side-encryption-key-management/).
```

Comment:
Do we need to do that if we're using Atlas? I anticipate Atlas will have a simpler solution.
Can we have a comment/section explaining what people using Atlas need to do?

Reply:
That's a good point, I'll have to ask the MongoDB team on that front. I have never run CSFLE in production, so I don't know the answer to this right now.

Reply:
In that case we can merge this as is and add a change if needed when we get a response. This is already an improvement to the docs.